tracing-subscriber/src/fmt/time.rs

```diff
@@ -268,3 +268,217 @@ where
     timer.format_time(writer)?;
     write!(writer, " ")
 }
+
+/// A date/time type which exists primarily to convert `SystemTime` timestamps into an ISO 8601
+/// formatted string.
+///
+/// Yes, this exists. Before you have a heart attack, understand that the meat of this is musl's
+/// [`__secs_to_tm`][1] converted to Rust via [c2rust][2] and then cleaned up by hand. All existing
+/// `strftime`-like APIs I found were unable to handle the full range of timestamps representable
+/// by `SystemTime`, including `strftime` itself, since tm.tm_year is an int.
+///
+/// TODO: figure out how to properly attribute the MIT licensed musl project.
+///
+/// [1] http://git.musl-libc.org/cgit/musl/tree/src/time/__secs_to_tm.c
+/// [2] https://c2rust.com/
+///
+/// This is directly copy-pasted from https://github.com/danburkert/kudu-rs/blob/c9660067e5f4c1a54143f169b5eeb49446f82e54/src/timestamp.rs#L5-L18
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct DateTime {
```

Comment:
This isn't part of the public API --- it's only used internally by the `SystemTime` formatter. Therefore, it doesn't need to be `pub`:
```suggestion
struct DateTime {
```